src/arkode/arkode_lsrkstep.c

```diff
@@ -1921,6 +1976,17 @@ int lsrkStep_TakeStepSSP104(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPt
   Xvecs[2] = ark_mem->tempv3;
 
   retval = N_VLinearCombination(3, cvals, Xvecs, ark_mem->ycur);
+  if (retval != 0)
+  {
+    SUNLogInfo(ARK_LOGGER, "end-stage",
+               "status = failed vector op, retval = %i", retval);
+    return ARK_VECTOROP_ERR;
+  }
+
+  SUNLogInfo(ARK_LOGGER, "end-stage", "status = success");
+  SUNLogExtraDebugVec(ARK_LOGGER, "updated solution", ark_mem->ycur, "ycur(:) =");
+  SUNLogExtraDebugVec(ARK_LOGGER, "embedded solution", ark_mem->tempv1,
```

Comment:
I can't highlight enough lines to make the "suggestion" correctly, but this and the following line need to be moved just after the `N_VLinearSum` call in the following `if (!ark_mem->fixedstep)` block.